packages/wonder-stuff-core/src/__tests__/values.flowtest.js

```diff
@@ -0,0 +1,36 @@
+// @flow
+import {values} from "../values.js";
+
+{
+    // should type returned array element with union of value types of passed object
+    const obj1 = {
+        a: 1,
+        b: "2",
+        c: [3, 4],
+    };
+    const obj1Values = values(obj1);
+
+    // This works because the variable is typed to all the possible value types.
+    const _: number | string | Array<number> = obj1Values[0];
+
+    // This errors because the variable is only typed to string, but the value
+    // could be a number, string, or array of numbers.
+    // $FlowExpectedError[incompatible-type]
+    const __: string = obj1Values[1];
+}
+
+{
+    // should error if passed object values do not match parameterized type
+    const obj2 = {
+        a: 1,
+        b: "2",
+    };
+
+    // This errors because the return type of values() is not Array<number>
+    // $FlowExpectedError[incompatible-type]
+    const _: Array<number> = values(obj2);
+
+    // This errors because the object does not have values that are all numbers.
+    // $FlowExpectedError[incompatible-call]
+    const __ = values<number>(obj2);
+}
```

Comment:
I'd be interested to see what type Flow infers for the array if the object has no properties. It also might be worth including a flowtest that uses an object-as-map type like `{|[string]: number|}`. I think Flow will infer struct-like types for the objects in the current tests.

Reply:
What outcome would these tests be catching/looking for? Generally these tests are to assert the types are working and to give us a signal if they break. What specifically would the test case be for in these examples?

Reply:
I expect `values` to accept objects-as-maps, and infer the correct output type, so it seems valuable to have a flowtest  ensuring that works. For the empty-object case, I was mostly just curious what the return type would be. I agree it doesn't seem very useful to have a test for that.